recipes/dataframe/all/conanfile.py

```diff
@@ -54,48 +75,24 @@ def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
-        compiler = str(self.settings.compiler)
-        compiler_version = tools.Version(self.settings.compiler.version)
-
-        if compiler == "Visual Studio" and self.options.shared:
+    def validate(self):
+        if self._is_msvc and self.options.shared and tools.Version(self.version) <= "1.19.0":
             raise ConanInvalidConfiguration(
-                "Could not support this specific configuration. Try static."
+                "dataframe {} doesn't support shared lib with Visual Studio".format(self.version)
             )
```

Comment:
should be fixed in the next release, so I've added this condition to not raise in this case: https://github.com/hosseinmoein/DataFrame/pull/153